p2p/src/protocol/cbfmgr.rs

```diff
@@ -501,7 +509,13 @@ impl<F: Filters, U: SyncFilters + Events + Wakeup + Disconnect, C: Clock> Filter
         if range.is_empty() {
             return Err(GetFiltersError::InvalidRange);
         }
-        assert!(*range.end() <= self.filters.height());
+
+        let current_header_height = self.filters.height();
+        if *range.end() > current_header_height {
+            return Err(GetFiltersError::HeaderNotDownloaded {
+                current_header_height,
+            });
+        }
```

Comment:
We already have the `InvalidRange` error for this that we could use.

Reply:
Hmmm, do we also return the `current_header_height` too?
```rust
            return Err(GetFiltersError::InvalidRange {
                current_header_height,
            });
```

Reply:
>Hmmm, do we also return the current_header_height too?

Yes, this is useful from a UI point of view. The lightning network node at the moment are not smart enough and continue asking for the next block also if the `InvalidRange` occurse

Reply:
We can return the tip yes, just call it `tip`

Reply:
Done. Thanks @cloudhead @vincenzopalazzo